### PR TITLE
feat(prover): add unit test for circuits e2e test (used in test_zkp project)

### PR DESCRIPTION
**Pull request by amoylan2**
### Purpose or design rationale of this PR

feat(prover): add unit test for circuits e2e test (used in test_zkp project)

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [x] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Comment by amoylan2** [comment]
Currently, I'm testing this in this machine ` | idc-us-node-19  | 192.168.1.169`
The working directory is  `/home/ubuntu/mengran/prover_unittests`
And just running ./start.sh to see the test result.

The test failed with an error `Floating point exception (core dumped)`. The full log is in test.log
I have talked to @lispc , who will check it next week.

File: prover/src/zk_circuits_handler/bernoulli.rs [hunk 1]
```diff
@@ -70,34 +69,50 @@ impl BaseCircuitsHandler {
...
+        serde_json::to_string(&chunk_proof).map_err(|e| anyhow::anyhow!(e))
+    }
 
+    fn gen_batch_proof_raw(
+        &self,
+        chunk_hashes_proofs: Vec<(ChunkHash, ChunkProof)>,
+    ) -> Result<BatchProof> {
         if let Some(prover) = self.batch_prover.as_ref() {
+            let chunk_proofs: Vec<ChunkProof> =
+                chunk_hashes_proofs.iter().map(|t| t.1.clone()).collect();
```
Comment (georgehao) [inline prover/src/zk_circuits_handler/bernoulli.rs:89]: why need `clone` here?

File: prover/src/zk_circuits_handler/bernoulli.rs [hunk 2]
```diff
@@ -195,3 +210,180 @@ impl CircuitsHandler for BaseCircuitsHandler {
...
+    use prover::utils::chunk_trace_to_witness_block;
+    use std::{path::PathBuf, sync::LazyLock};
+
+    #[ctor::ctor]
+    fn init() {
+        crate::utils::log_init(None);
+        log::info!("logger initialized");
+    }
+
+    static DEFAULT_WORK_DIR: &str = "/assets";
```
Comment (georgehao) [inline prover/src/zk_circuits_handler/bernoulli.rs:229]: why define a static constant?

File: prover/src/zk_circuits_handler/bernoulli.rs (hunk 1 above)
Comment (amoylan2) [reply prover/src/zk_circuits_handler/bernoulli.rs:89]: the chunk_hashes_proofs will be passed to gen_agg_evm_proof function, so the chunk_proofs part need to be cloned here

File: prover/src/zk_circuits_handler/bernoulli.rs [hunk 3]
```diff
@@ -195,3 +210,180 @@ impl CircuitsHandler for BaseCircuitsHandler {
...
+    static WORK_DIR: LazyLock<String> = LazyLock::new(|| {
+        std::env::var("BERNOULLI_TEST_DIR")
+            .unwrap_or(String::from(DEFAULT_WORK_DIR))
+            .trim_end_matches('/')
+            .to_string()
+    });
+    static PARAMS_PATH: LazyLock<String> =
+        LazyLock::new(|| format!("{}/test_params", WORK_DIR.clone()));
```
Comment (georgehao) [inline prover/src/zk_circuits_handler/bernoulli.rs]:
```suggestion
        LazyLock::new(|| format!("{}/test_params", *WORK_DIR));
```
Is it better?

**Comment by georgehao** [comment]
Anyhow, I think you should add the integration test which you have deleted associated golang prover

**Comment by lispc** [comment]
will check this afternoon

File: prover/src/zk_circuits_handler/bernoulli.rs (hunk 3 above)
Comment (amoylan2) [reply prover/src/zk_circuits_handler/bernoulli.rs]: done

File: prover/src/zk_circuits_handler/bernoulli.rs [hunk 4]
```diff
@@ -70,34 +69,50 @@ impl BaseCircuitsHandler {
...
         }
         unreachable!("please check errors in proof_type logic")
     }
 
+    fn gen_batch_proof(&self, task: &crate::types::Task) -> Result<String> {
+        log::info!("[circuit] gen_batch_proof for task {}", task.id);
+        let chunk_hashes_proofs: Vec<(ChunkHash, ChunkProof)> =
+            self.gen_chunk_hashes_proofs(task)?;
+        let batch_proof = self.gen_batch_proof_raw(chunk_hashes_proofs)?;
+        serde_json::to_string(&batch_proof).map_err(|e| anyhow::anyhow!(e))
```
Comment (lispc) [inline prover/src/zk_circuits_handler/bernoulli.rs]: will `Ok(serde_json::to_string(&batch_proof)?)` work?

**Review by lispc — DISMISSED** [review]

File: prover/src/zk_circuits_handler/bernoulli.rs (hunk 4 above)
Comment (amoylan2) [reply prover/src/zk_circuits_handler/bernoulli.rs]: Yes, it works. Learned a new tip

Comment (amoylan2) [reply prover/src/zk_circuits_handler/bernoulli.rs]: have submitted a new commit fixing this

**Review by lispc — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

**Pull request merged**